sdk/python/kfp/components/_python_op.py

```diff
@@ -113,24 +86,61 @@ def annotation_to_type_struct(annotation):
                 type=type_struct,
             )
             outputs.append(output_spec)
-            extra_output_names.append(field_name)
-            arguments.append(OutputPathPlaceholder(field_name))
     elif signature.return_annotation is not None and signature.return_annotation != inspect.Parameter.empty:
```

Comment:
What if it is a tuple, should each output be extracted? This looks like it only extract multiple outputs when it is a NamedTuple.

Reply:
How does the tuple output _annotation_ look like?
We need to know the output signature - the names of the outputs (and their types).

Reply:
What if it is a dict? e.g. https://github.com/kubeflow/pipelines/blob/7e29b431bc7696b5c5468dc6d714d18858d63ecd/sdk/python/tests/dsl/component_tests.py#L33
I think this function can later be used in the dsl as well.

Reply:
Oh. I see. I agree that we should improve the type support here. I just want to keep refactorings separate from changes. It's much easier to track and review thing that way.
I'll expand this function in another PR.